src/selectors/entities/channels.js

```diff
@@ -525,30 +550,34 @@ export const getSortedUnreadChannelIds = createIdsSelector(
             }
 
             return c;
-        }).filter((c) => c).sort((a, b) => {
```

Comment:
Filter here is used for removing deleted channels re: line 544:546 

```
if (c.delete_at !== 0) {
     return false;
}
```

Reply:
Really good catch. I didn't see that before. We should actually make it so that a filter is done to remove deleted channels before the map call instead of returning false sometimes. Having mixed return types from the map statement is not something I expected at all, and it definitely causes problems like the one you caught

Reply:
I am not a fan of this logic here. I will let miguelespinoza decide if he wants to fix this or leave it as two maps. 
Having a reduce would have done the job rather than mixed return types and two `maps`

I recently almost broke this https://github.com/mattermost/mattermost-redux/pull/680.
and thought about refactoring it myself so it obvious on first look so we don't run into issues like this but coudn't make the time.

Reply:
A reduce would do it as well, but this only needs one map
```javascript
const allUnreadChannels = unreadIds.filter((id) => channels[id] && channels[id].delete_at > 0).map(...);
```

Reply:
👍 Aah ya that is more readable. I tend to resort to reduce whenever i need to filter and iterate.

Reply:
Nice, thanks for the feedback @hmhealey and nice catch @sudheerDev.  I'll address this hopefully today.  If not for sure tomorrow. I'm glad you found that catch @sudheerDev 👍